### ISSUE TITLE
Recent docker forces usage of new Docker Compose V2

### ISSUE THREAD

**Issue by andileco** [lando/lando#3075]
Recently got a new work computer and found out it had an M1 chip. I can't get Lando working and get errors related to "services":

**"lando rebuild -y" (or destroy) gives me:** 

services.appserver_nginx.env_file must be a string

**"lando start" gives me:** 

services.ca.ports must be a list

"lando info" gives me:

[ { service: 'appserver_nginx',
    urls: [],
    type: 'nginx',
    healthy: true,
    managed: true,
    webroot: 'web',
    config:
     { vhosts: '/Users/me/.lando/config/pantheon/drupal8.conf.tpl',
       php: '/Users/me/.lando/config/pantheon/php.ini',
       server: '/Users/me/.lando/config/pantheon/nginx.conf.tpl' },
    version: '1.16',
    meUser: 'www-data',
    hasCerts: true,
    hostnames: [ 'appserver_nginx.mysite.internal' ] },
  { service: 'appserver',
    urls: [],
    type: 'php',
    healthy: true,
    via: 'nginx:1.16',
    served_by: 'appserver_nginx',
    webroot: 'web',
    config:
     { php: '/Users/me/.lando/config/pantheon/php.ini',
       server: '/Users/me/.lando/config/pantheon/nginx.conf.tpl',
       vhosts: '/Users/me/.lando/config/pantheon/drupal8.conf.tpl' },
    version: '7.4',
    meUser: 'www-data',
    hasCerts: false,
    hostnames: [ 'appserver.mysite.internal' ] },
  { service: 'database',
    urls: [],
    type: 'mysql',
    healthy: true,
    internal_connection: { host: 'database', port: '3306' },
    external_connection: { host: '127.0.0.1', port: true },
    healthcheck: 'bash -c "[ -f /bitnami/mysql/.mysql_initialized ]"',
    creds: { database: 'pantheon', password: 'pantheon', user: 'pantheon' },
    config: { database: '/Users/me/.lando/config/pantheon/mysql.cnf' },
    version: '5.7',
    meUser: 'www-data',
    hasCerts: false,
    hostnames: [ 'database.mysite.internal' ] } ]

**.lando.yml file:**

name: my-site
recipe: pantheon
config:
  framework: drupal8
  site: my-site
  id: [redacted]
  '#' Disable the SOLR index
  index: false
  '#' Disable the VARNISH edge
  edge: false
  '#' Disable the REDIS cache
  cache: false
services:
  appserver:
    scanner: false
  database:
    type: mysql
    portforward: true
    creds:
      user: pantheon
      password: pantheon
      database: pantheon
    healthcheck: null

**Comment by yobottehg** [comment]
I also get this errors without an M1 chip. Perhaps some fresh released containers?

**Comment by pirog** [comment]
Next version of Lando will force usage of the older docker-compose until we officially support the newer one. This will happen regardless of whether you've checked the box in Docker Desktop or not.

**Comment by Tbaile** [comment]
In the Arch Linux community repository, docker-compose has been officially updated to 2.0.0.
There's anything to be done to implement support for the new version of compose? The [docs](https://docs.docker.com/compose/cli-command-compatibility/) that define the compatibility between the two versions suggests that the changes are very little and none of them should be gamebreaking for lando.

**Comment by pirog** [comment]
We will support `2.x` at some point

**Comment by Tbaile** [comment]
I've browsed the code a little, found out that the only changes actually needed for it to work is to change (based on the docker-compose version detected) how compose commands are built. If the implementation is somewhat clean I'll send this through a PR.

**Comment by pirog** [comment]
I don’t think we want to just wholesale force move everything and everyone
over to V2 especially considering it’s highly probable, despite claims that
it is a drop in replacement that “should” just work, that it’s not going to
just work.

We probably want to make this configurable to the user so they can opt in
to using a “beta” of v2 if they want. This way we can determine when we
think it’s ready for wholesale use without compromising the stability for
everyone.

**Comment by Tbaile** [comment]
Absolutely, the V2 is currently a release candidate and not ready for production as documentation states, I've tried with a bunch of projects locally and everything works, I've surely not tested every use of application of this new version.
Lando could show a warning for a "beta release" of compose and just work by changing, based on the version of compose detected, how compose.js generates the commands. Then we'll have what we need to change in order for lando to support the newer version before this drops on any major platform.

**Comment by reynoldsalec** [comment]
@Tbaile does that mean that most users could be running Docker Compose V2 and won't experience any issues without any changes to Lando?

Hoping we can close this out... :)

**Comment by Tbaile** [comment]
Yes, the issue can be closed.
I don't think there's the need to run a check within Lando since very few people will encounter this issue.

TLDR:
If the docker-compose version you run is _exactly_ the 2.0.0, you wont be able to run Lando at all. You need the version 2.0.1 or above that integrates the "conversion" script from the previous commands to the new plugin.

**Comment by reynoldsalec** [comment]
Awesome, thanks @Tbaile, closing this out.
